source/linux/Foundational_Components/U-Boot/Apps-Splash-Screen.rst

```diff
@@ -10,247 +10,62 @@ U-Boot Splash Screen
-called am3355x-fb.c. It makes calls to routines in board.c to set up the
-LCDC and frame buffer. To use it:
+For AM335x, splash screen is enabled in u-boot which uses the LCDC driver.
+On AM335x evm, the LCD backlight is controlled using enhanced Auxiliary(eCAP0) PWM.
+To use this, enable following configurations in configs/am335x_evm_defconfig.
 
-Either create a new defconfig in the configs directory or just add
-SPLASH to CONFIG\_SYS\_EXTRA\_OPTIONS. In this example the
-am335x\_evm\_defconfig is copied into a new one called
-am335x\_evm\_splash\_defconfig.
+.. code-block:: kconfig
 
-.. code-block:: text
+    CONFIG_SPLASH_SCREEN=y
+    CONFIG_SPLASH_SOURCE=y
+    CONFIG_SPLASH_SCREEN_ALIGN=y
+    CONFIG_HIDE_LOGO_VERSION=y
+    CONFIG_AM335X_LCD=y
+    CONFIG_VIDEO_BMP_GZIP=y
+    CONFIG_BMP_24BPP=y
+    CONFIG_BMP_32BPP=y
+    CONFIG_BMP=y
+    CONFIG_VIDEO=y
+    CONFIG_CMD_BMP=y
+    CONFIG_SYSCON=y
+    CONFIG_REGMAP=y
+    CONFIG_CLK_CCF=y
+    CONFIG_CLK_TI_AM3_DPLL=y
+    CONFIG_CLK_TI_DIVIDER=y
+    CONFIG_CLK_TI_MUX=y
+    CONFIG_DM_PWM=y
+    CONFIG_BACKLIGHT_PWM=y
+    CONFIG_PWM_TI_ECAP=y
 
-    CONFIG_TARGET_AM335X_EVM=y
-    CONFIG_SPL_STACK_R_ADDR=0x82000000
-    CONFIG_DEFAULT_DEVICE_TREE="am335x-evm"
-    CONFIG_SPL=y
-    CONFIG_SPL_STACK_R=y
-    CONFIG_SYS_EXTRA_OPTIONS="NAND,SPLASH"
-    CONFIG_HUSH_PARSER=y
-    CONFIG_AUTOBOOT_KEYED=y
+.. rubric:: Display Image using U-Boot command line
+   :name: Display Image
 
-In include/configs/am335x\_evm.h, add support for the splash screen,
-LCDC, and gzipped bitmaps.
+To test the display and video driver in U-Boot, run the following commands at U-Boot console:
 
-::
+.. code-block:: console
 
-    /* Splash scrren support */
-    #ifdef CONFIG_SPLASH
-    #define CONFIG_AM335X_LCD
-    #define CONFIG_LCD
-    #define CONFIG_LCD_NOSTDOUT
-    #define CONFIG_SYS_WHITE_ON_BLACK
-    #define LCD_BPP LCD_COLOR16
+   #To see all the files in your boot partition run
+   => ls mmc 0
 
-    #define CONFIG_VIDEO_BMP_GZIP
-    #define CONFIG_SYS_VIDEO_LOGO_MAX_SIZE  (1366*767*4)
-    #define CONFIG_CMD_UNZIP
-    #define CONFIG_CMD_BMP
-    #define CONFIG_BMP_16BPP
-    #endif
+   #To load image
+   => fatload mmc 0 $loadaddr ti_logo_414x97_32bpp.bmp.gz
 
-In arch/arm/cpu/armv7/am33xx/clock\_am33xx.c enable the LCDC clocks.
+   #To display image
+   =>  bmp display $loadaddr m m
 
-::
+The above command will display an image at the centre of the screen as below
 
-                    &cmrtc->rtcclkctrl,
-                    &cmper->usb0clkctrl,
-                    &cmper->emiffwclkctrl,
-                    &cmper->emifclkctrl,
-                    &cmper->lcdclkctrl,
-                    &cmper->lcdcclkstctrl,
-                    &cmper->epwmss2clkctrl,
-                    0
+.. Image:: /images/AM335x_evm_uboot_splash_screen.jpg
+   :width: 600
 
-In board.c add includes for mmc, fat, lcd, and the frame buffer.
+User can use the custom image to display the splash screen.
```

Comment:
```suggestion
The user may use any bitmap image as splash screen. The below command may be used to verify the image is valid.
```

Even this is a little vague. EFI boot splash on amd64 still has general requirements about image size, alpha channel usage, etc. There needs to be more information here.

Reply:
@sukrutb any comments on this?